packages/sdk/src/protocols/base_protocol.ts

```diff
@@ -1,15 +1,206 @@
+import type { Peer, PeerId } from "@libp2p/interface";
```

Comment:
just a comment, not connected to this PR:
I don't remember why we implemented this file, but I think we shouldn't have `SDK` for `base_protocol` as it is internal and only user facing API should be SDK 

let's discuss @waku-org/js-waku-developers

Reply:
why this core logic cannot be in the `bae_protocol` in the `core`?

Reply:
let's get back to it later in https://github.com/waku-org/js-waku/issues/2033